modules/server/src/network/index.ts

```diff
@@ -1,10 +1,93 @@
-import { NetworkAggregationInterface } from '@/config/types';
+import { buildClientSchema, getIntrospectionQuery, IntrospectionQuery } from 'graphql';
+import { fetchGql } from './gql';
+import { NetworkAggregationConfig, NetworkAggregationConfigInput } from './types';
 
-export const createSchemaFromNetworkConfig = ({
+const isGqlIntrospectionQuery = (input: any): input is IntrospectionQuery => {
+	return (input as IntrospectionQuery).__schema !== undefined;
+};
+
+type FetchRemoteSchemaResult = {
+	config: NetworkAggregationConfigInput;
+	introspectionResult: IntrospectionQuery | null;
+};
+
+/**
+ *
+ * @param config - network config from env
+ * @returns a promise containing network config and the introspection query result
+ *
+ * @throws Fetch failed error
+ *
+ * @throws JSON parse error
+ *
+ * @throws Unexpected data error eg. Not a valid introspectionQuery result
+ */
+const fetchRemoteSchema = async (
+	config: NetworkAggregationConfigInput,
+): Promise<FetchRemoteSchemaResult> => {
+	const { graphqlUrl } = config;
+	try {
+		/**
+		 * get full schema (needed for buildClientSchema) and convert json
+		 */
+		const response = await fetchGql({
+			url: graphqlUrl,
+			gqlRequest: { query: getIntrospectionQuery() },
+		});
+
+		const jsonData = await response.json();
+
+		if (jsonData && jsonData.data && isGqlIntrospectionQuery(jsonData.data)) {
+			return { config, introspectionResult: jsonData };
+		} else {
+			throw Error('response data unexpected');
+		}
+	} catch (error) {
+		console.log(`failed to retrieve schema from url: ${config.graphqlUrl}`);
+		console.error(error);
+		return { config, introspectionResult: null };
+	}
+};
+
+const fetchRemoteSchemas = async ({
+	networkConfig,
+}: {
+	networkConfig: NetworkAggregationConfigInput[];
+}): Promise<NetworkAggregationConfig[]> => {
+	// query remote notes
+	// type cast because rejected errors are handled
+	const networkQueries = (await Promise.allSettled(
+		networkConfig.map(async (config) => fetchRemoteSchema(config)),
+	)) as PromiseFulfilledResult<FetchRemoteSchemaResult>[];
```

Comment:
wow! there's a lot going on here! any chance this can be chunked out into separate steps?

Reply:
https://github.com/overture-stack/arranger/pull/877/commits/050bf5130ddc819d4a2d3a33b36d492f380276b1